clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -720,6 +721,12 @@ private void maybeOverrideEnableAutoCommit(Map<String, Object> configs) {
         }
     }
 
+    private void checkGroupRemoteAssignor() {
+        if (getString(GROUP_PROTOCOL_CONFIG).equals(GroupProtocol.CLASSIC.name().toLowerCase(Locale.ROOT)) && getString(GROUP_REMOTE_ASSIGNOR_CONFIG) != null && !getString(GROUP_REMOTE_ASSIGNOR_CONFIG).isEmpty()) {
+            throw new InvalidConfigurationException(GROUP_REMOTE_ASSIGNOR_CONFIG + " cannot be set when default group protocol (classic) is used.");
+        }
```

Comment:
There are some changes I think should be made here:

1. The `group.protocol` configuration is case-insensitive, so we can't assume it will be lower-case
2. The exception type used for configuration errors is `ConfigException` vs. `InvalidConfigurationException`

```suggestion
        if (getString(GROUP_PROTOCOL_CONFIG).equalsIgnoreCase(GroupProtocol.CLASSIC.name()) && getString(GROUP_REMOTE_ASSIGNOR_CONFIG) != null && !getString(GROUP_REMOTE_ASSIGNOR_CONFIG).isEmpty()) {
            throw new ConfigException(GROUP_REMOTE_ASSIGNOR_CONFIG + " cannot be set when " + GROUP_PROTOCOL_CONFIG + "=" + GroupProtocol.CLASSIC.name());
        }
```

Also, since the `CLASSIC` group protocol won't always be the default, I suggested a rewording of the error, but that's subjective, so I won't insist on it.